gnovm/pkg/doc/json_doc.go

```diff
@@ -232,14 +374,71 @@ func (d *Documentable) extractValueSpecs(pkg *doc.Package, specs []ast.Spec) []*
 
 // mustFormatNode calls format.Node and returns the result as a string.
 // Panic on error, which shouldn't happen since the node is a valid AST from pkgData.parseFile.
-func mustFormatNode(fset *token.FileSet, node any) string {
+// If source is true and the optional ast.File is given, then use it to get internal comments.
+func mustFormatNode(fset *token.FileSet, node any, source bool, file *ast.File) string {
+	if !source {
+		// Temporarily remove the Doc and Body so that it's not in the signature
+		switch n := node.(type) {
+		case *ast.FuncDecl:
+			saveDoc := n.Doc
+			saveBody := n.Body
+			defer func() {
+				n.Doc = saveDoc
+				n.Body = saveBody
+			}()
+			n.Doc = nil
+			n.Body = nil
```

Comment:
what about something like `node = &ast.FuncDecl{...}`?

Reply:
See commit https://github.com/gnolang/gno/pull/3931/commits/1ffb7adb3ecc1847dbbb1952b723466a49333deb . I agree that this looks cleaner but I worry that it's not forward compatible. (What if they add a new field to `ast.FuncDecl` which isn't copied here?)